genomedata/__init__.py

```diff
@@ -203,23 +188,12 @@ def __getitem__(self, name):
 
         """
         assert self.isopen
-        try:
-            # memoization
-            return self.open_chromosomes[name]
-        except KeyError:
-            pass
 
         try:
-            if self._isfile:
-                res = Chromosome(self.h5file, where="/" + name)
-            else:
-                res = Chromosome._fromfilename(
-                    self._path.joinpath(name + SUFFIX),
-                    *self.args, **self.kwargs)
+            res = self._chromosomes[name]
         except (IOError, NoSuchNodeError):
```

Comment:
`NoSuchNodeError` is a fairly specific implementation issue. I think this code should be in the list object instead

Reply:
The list objects will raise the exception with the same error message here